core/search/disjunction.hpp

```diff
@@ -104,100 +89,77 @@ class min_match_buffer<0> {
     assert(false);
     return 1;
   }
-  uint32_t min_match_count() const noexcept {
-    return 1;
-   }
-}; // min_match_buffer
+  uint32_t min_match_count() const noexcept { return 1; }
+};
 
 class score_buffer {
  public:
-  score_buffer(const order::prepared& ord, size_t size)
-    : bucket_size_(ord.score_size()),
-      buf_size_(bucket_size_*size),
-      buf_(ord.empty() ? nullptr : new byte_type[buf_size_]) {
+  score_buffer(size_t num_buckets, size_t size)
+      : bucket_size_{num_buckets * sizeof(score_t)},
+        buf_size_{bucket_size_ * size},
+        buf_{!buf_size_ ? nullptr : new byte_type[buf_size_]} {
     if (buf_) {
       std::memset(data(), 0, this->size());
     }
   }
 
-  byte_type* get(size_t i) noexcept {
-    assert(!buf_ || bucket_size_*i < buf_size_);
-    return buf_.get() + bucket_size_*i;
+  score_t* get(size_t i) noexcept {
+    assert(!buf_ || bucket_size_ * i < buf_size_);
+    return reinterpret_cast<score_t*>(buf_.get() + bucket_size_ * i);
   }
 
-  byte_type* data() noexcept {
-    return buf_.get();
-  }
+  score_t* data() noexcept { return reinterpret_cast<score_t*>(buf_.get()); }
 
-  size_t size() const noexcept {
-    return buf_size_;
-  }
+  size_t size() const noexcept { return buf_size_; }
 
-  size_t bucket_size() const noexcept {
-    return bucket_size_;
-  }
+  size_t bucket_size() const noexcept { return bucket_size_; }
 
  private:
   size_t bucket_size_;
   size_t buf_size_;
   std::unique_ptr<byte_type[]> buf_;
-}; // score_buffer
+};  // score_buffer
```

Comment:
As a suggestion - maybe use here a small vector as `buf_`? An later even fine tune inline size for different disjunction types.